src/makeInfinityAdder.js

```diff
@@ -4,9 +4,22 @@
  *
  * @return {function}
  */
+let sum = 0;
 
 function makeInfinityAdder() {
-  // write code here
+  return (number) => {
+    if (number) {
+      sum += number;
+
+      return makeInfinityAdder();
```

Comment:
here we can write - **return makeInfinityAdder(x + number);** without sum += number